packages/nuxt/src/app/composables/asyncData.ts

```diff
@@ -212,7 +212,11 @@ export function useAsyncData<
         // If this request is cancelled, resolve to the latest request.
         if ((promise as any).cancelled) { return nuxt._asyncDataPromises[key] }
 
-        asyncData.error.value = error
+        const nuxtError = createError(error)
+
+        nuxt.payload._errors[key] = nuxtError
+
+        asyncData.error.value = nuxtError as DataE
```

Comment:
do we need to do both of these assignments? `asyncData.error` should already be referencing `nuxt.payload._errors[key]` as it was created with `toRef`....

```suggestion
        asyncData.error.value = createError(error) as DataE
```

Reply:
If the following destructuring doesn't break reactivity, I'll make an amendment.
https://github.com/nuxt/nuxt/blob/4930d273c9b4a4029bf6dcc51c1f3e99b65adfc9/packages/nuxt/src/app/composables/asyncData.ts#L171

Reply:
Thank you! We could add a test to ensure it.